osu.Game/Storyboards/Drawables/DrawableStoryboard.cs

```diff
@@ -142,14 +142,32 @@ public StoryboardResourceLookupStore(Storyboard storyboard, RealmAccess realm, G
             public void Dispose() =>
                 realmFileStore.Dispose();
 
-            public byte[] Get(string name) =>
-                realmFileStore.Get(storyboard.GetStoragePathFromStoryboardPath(name));
+            public byte[] Get(string name)
+            {
+                string? storagePath = storyboard.GetStoragePathFromStoryboardPath(name);
 
-            public Task<byte[]> GetAsync(string name, CancellationToken cancellationToken = new CancellationToken()) =>
-                realmFileStore.GetAsync(storyboard.GetStoragePathFromStoryboardPath(name), cancellationToken);
+                return string.IsNullOrEmpty(storagePath)
+                    ? null!
+                    : realmFileStore.Get(storagePath);
+            }
 
-            public Stream GetStream(string name) =>
-                realmFileStore.GetStream(storyboard.GetStoragePathFromStoryboardPath(name));
+            public Task<byte[]> GetAsync(string name, CancellationToken cancellationToken = new CancellationToken())
+            {
+                string? storagePath = storyboard.GetStoragePathFromStoryboardPath(name);
+
+                return string.IsNullOrEmpty(storagePath)
+                    ? Task.FromResult<byte[]>(null!)
+                    : realmFileStore.GetAsync(storagePath, cancellationToken);
+            }
```

Comment:
The `null!`s here are slightly annoying. You would probably very wisely say to do this:

```diff
diff --git a/osu.Game/Storyboards/Drawables/DrawableStoryboard.cs b/osu.Game/Storyboards/Drawables/DrawableStoryboard.cs
index c2a58d46ef..631475cce8 100644
--- a/osu.Game/Storyboards/Drawables/DrawableStoryboard.cs
+++ b/osu.Game/Storyboards/Drawables/DrawableStoryboard.cs
@@ -128,7 +128,7 @@ private void updateLayerVisibility()
                 layer.Enabled = passing ? layer.Layer.VisibleWhenPassing : layer.Layer.VisibleWhenFailing;
         }
 
-        private class StoryboardResourceLookupStore : IResourceStore<byte[]>
+        private class StoryboardResourceLookupStore : IResourceStore<byte[]?>
         {
             private readonly IResourceStore<byte[]> realmFileStore;
             private readonly Storyboard storyboard;
@@ -142,21 +142,21 @@ public StoryboardResourceLookupStore(Storyboard storyboard, RealmAccess realm, G
             public void Dispose() =>
                 realmFileStore.Dispose();
 
-            public byte[] Get(string name)
+            public byte[]? Get(string name)
             {
                 string? storagePath = storyboard.GetStoragePathFromStoryboardPath(name);
 
                 return string.IsNullOrEmpty(storagePath)
-                    ? null!
+                    ? null
                     : realmFileStore.Get(storagePath);
             }
 
-            public Task<byte[]> GetAsync(string name, CancellationToken cancellationToken = new CancellationToken())
+            public Task<byte[]?> GetAsync(string name, CancellationToken cancellationToken = new CancellationToken())
             {
                 string? storagePath = storyboard.GetStoragePathFromStoryboardPath(name);
 
                 return string.IsNullOrEmpty(storagePath)
-                    ? Task.FromResult<byte[]>(null!)
+                    ? Task.FromResult<byte[]?>(null)
                     : realmFileStore.GetAsync(storagePath, cancellationToken);
             }
 

```

But the fun part is that at that point, there is a .NET library NRT fail:

![image](https://github.com/ppy/osu/assets/20418176/799ae06d-3485-4e46-b7d3-e1dfeb752fa6)

and to make it shut up, you have to do this abomination:

![1695148291](https://github.com/ppy/osu/assets/20418176/6e820337-fd12-4c41-835b-3833b688e41d)

Yes you're reading that right, that's two shut-up operator occurrences. So I decided I'd rather leave well alone.